vscode/microsoft-kiota/src/steps.ts

```diff
@@ -161,15 +161,19 @@ export async function generateSteps(existingConfiguration: Partial<GenerateState
                 return inputClientClassName;
             case 'plugin':
                 return inputPluginName;
-            case 'apimanifest':
-                return inputManifestName;
+            case 'other':
```

Comment:
This has broken deeplinking generation. If I pass in deeplink params as 
kind=Plugin, type=ApiManifest, source=ttk, name=Githubv2RestAPI, an error will be thrown for unknown generation type[state.generationType=apimanifest/openai no longer exist](https://github.com/microsoft/kiota/blob/main/vscode/microsoft-kiota/src/steps.ts#L167). In transformToGenerationConfig we should map case openai and apimanifest correcty to `generationConfig.generationType="other"` in [ transformToGenerationConfig](https://github.com/microsoft/kiota/blob/main/vscode/microsoft-kiota/src/utilities/deep-linking.ts#L25) so that if case/deeplink type "openai" or "apimanifest" are passed, they correctly transform to generationType Other in the generatesteps.

Reply:
This is a nice catch @fey101 ! 

I've added a few checks to ensure that it works on deep linking